operator/pkg/ciliumidentity/namespace.go

```diff
@@ -3,8 +3,61 @@
 
 package ciliumidentity
 
-import "github.com/cilium/cilium/pkg/k8s/resource"
+import (
+	"context"
+
+	ciliumio "github.com/cilium/cilium/pkg/k8s/apis/cilium.io"
+	"github.com/cilium/cilium/pkg/k8s/resource"
+	slimcorev1 "github.com/cilium/cilium/pkg/k8s/slim/k8s/api/core/v1"
+	"github.com/cilium/cilium/pkg/labels"
+	"github.com/cilium/cilium/pkg/labelsfilter"
+	"github.com/cilium/cilium/pkg/logging/logfields"
+	"github.com/cilium/cilium/pkg/policy"
+)
 
 func nsResourceKey(namespace string) resource.Key {
 	return resource.Key{Name: namespace}
 }
+
+func (c *Controller) processNamespaceEvents(ctx context.Context) error {
+	for event := range c.namespace.Events(ctx) {
+		switch event.Kind {
+		case resource.Upsert:
+			c.logger.Debug("Got Upsert Namespace event",
+				logfields.K8sNamespace, event.Key.String())
+
+			c.onNamespaceUpsert(event.Object)
+		}
+		event.Done(nil)
+	}
+	return nil
+}
+
+// onNamespaceUpsert sends namespace to Cilium Identity for reconciliation if
+// the namespace labels are changed.
+func (c *Controller) onNamespaceUpsert(ns *slimcorev1.Namespace) {
+	newLabels := getNamespaceLabels(ns)
+
+	oldIdLabels := c.oldNSSecurityLabels[ns.Name]
+	newIdLabels, _ := labelsfilter.Filter(newLabels)
+
+	// Do not perform any other operations if labels did not change.
+	if oldIdLabels.DeepEqual(&newIdLabels) {
+		return
+	}
+
+	c.oldNSSecurityLabels[ns.Name] = newIdLabels
+
+	if err := c.reconciler.reconcileNamespace(nsResourceKey(ns.Name)); err != nil {
+		c.logger.Error("Failed to process namespaces changes", logfields.Error, err)
```

Comment:
[minor] suggest making the log message closer to the code to make it easier to maintain: 

c.logger.Error("reconcileNamespace() failed", "Name", ns.Name, logfields.Error, err)